fluidkeys/secretsend.go

```diff
@@ -104,6 +102,56 @@ https://download.fluidkeys.com#` + recipientEmail + `
 	return 0
 }
 
+func getSecretFromFile(filename string) (*string, error) {
+	if fileExists(filename) {
+		secretData, err := ioutil.ReadFile(filename)
+		if err != nil {
+			return nil, fmt.Errorf("error reading file: " + err.Error())
+		}
+		secret := string(secretData)
+		out.Print("---\n")
+		out.Print(secret)
+		out.Print("---\n\n")
+
+		prompter := interactiveYesNoPrompter{}
+
+		if prompter.promptYesNo("Send "+filename+"?", "y", nil) {
+			return &secret, nil
+		}
+		return nil, nil
+	}
+	return nil, fmt.Errorf("couldn't find file " + filename)
+}
+
+func fileExists(filename string) bool {
+	if _, err := os.Stat(filename); err == nil {
+		return true
+	} else if os.IsNotExist(err) {
+		return false
+	} else {
+		log.Panic()
```

Comment:
What does this case mean? How does it happen?

Reply:
I got into a bit of a rabbit hole with this, having read https://stackoverflow.com/questions/12518876/how-to-check-if-a-file-exists-in-go

I now think that a better approach is to completely abandon testing if this file exists at this point in the code, as it's racy: the file may exist when we test, but may not when we go to open it. Instead I think we should simply try to open the file and if it fails, output the error we get.

Reply:
Cool.

In that case, definitely put something inside the `Panic()` so the user doesn't just have a mystery crash with no message :)